pkg/metaserver/spd/manager.go

```diff
@@ -19,214 +19,125 @@ package spd
+	// ServiceBusinessPerformanceLevel returns the service business performance level for the given pod
+	ServiceBusinessPerformanceLevel(ctx context.Context, pod *v1.Pod) (PerformanceLevel, error)
 
-type ServiceProfileManager interface {
-	// GetSPD get spd for given pod
-	GetSPD(ctx context.Context, pod *v1.Pod) (*workloadapis.ServiceProfileDescriptor, error)
+	// ServiceBusinessPerformanceScore returns the service business performance score for the given pod
+	// The score is in range [0, 100]
+	ServiceBusinessPerformanceScore(ctx context.Context, pod *v1.Pod) (float64, error)
 
-	// Run async loop to clear unused spd
+	// Run starts the service profiling manager
 	Run(ctx context.Context)
 }
 
-type spdManager struct {
-	started *atomic.Bool
-	mux     sync.Mutex
-
-	client            *client.GenericClientSet
-	emitter           metrics.MetricEmitter
-	cncFetcher        cnc.CNCFetcher
-	checkpointManager checkpointmanager.CheckpointManager
-	getPodSPDNameFunc GetPodSPDNameFunc
-
-	ServiceProfileCacheTTL time.Duration
+type DummyServiceProfilingManager struct{}
 
-	// spdCache is a cache of namespace/name to current target spd
-	spdCache *Cache
+func (d *DummyServiceProfilingManager) ServiceBusinessPerformanceLevel(_ context.Context, _ *v1.Pod) (PerformanceLevel, error) {
+	return PerformanceLevelPerfect, nil
 }
 
-// NewSPDManager creates a spd manager to implement ServiceProfileManager
-func NewSPDManager(clientSet *client.GenericClientSet, emitter metrics.MetricEmitter,
-	cncFetcher cnc.CNCFetcher, conf *pkgconfig.Configuration) (ServiceProfileManager, error) {
-	checkpointManager, err := checkpointmanager.NewCheckpointManager(conf.CheckpointManagerDir)
-	if err != nil {
-		return nil, fmt.Errorf("failed to initialize checkpoint manager: %v", err)
-	}
+func (d *DummyServiceProfilingManager) ServiceBusinessPerformanceScore(_ context.Context, _ *v1.Pod) (float64, error) {
+	return 100, nil
+}
 
-	m := &spdManager{
-		started:                atomic.NewBool(false),
-		client:                 clientSet,
-		emitter:                emitter,
-		checkpointManager:      checkpointManager,
-		cncFetcher:             cncFetcher,
-		ServiceProfileCacheTTL: conf.ServiceProfileCacheTTL,
-	}
+func (d *DummyServiceProfilingManager) Run(_ context.Context) {}
 
-	m.getPodSPDNameFunc = util.GetPodSPDName
-	m.spdCache = NewSPDCache(checkpointManager, defaultClearUnusedSPDPeriod)
+var _ ServiceProfilingManager = &DummyServiceProfilingManager{}
 
-	return m, nil
+type serviceProfilingManager struct {
+	fetcher SPDFetcher
 }
 
-func (s *spdManager) GetSPD(ctx context.Context, pod *v1.Pod) (*workloadapis.ServiceProfileDescriptor, error) {
-	spdName, err := s.getPodSPDNameFunc(pod)
+func NewServiceProfilingManager(clientSet *client.GenericClientSet, emitter metrics.MetricEmitter,
+	cncFetcher cnc.CNCFetcher, conf *pkgconfig.Configuration) (ServiceProfilingManager, error) {
+	fetcher, err := NewSPDFetcher(clientSet, emitter, cncFetcher, conf)
```

Comment:
You can implement custom ServiceProfilingManager by using NewSPDFetcher also, and we doesn't want someone use spd fetcher directly, because spd just one implement of service profiling manager